### ISSUE TITLE
[Android] Cannot read property 'className' of undefined

### ISSUE THREAD

**Issue by manijak** [NativeScript/theme#164]
nativescript-cli: 6.0.2
tns-core-modules: 6.0.4
theme-core: 2.0.5
platform-android: 6.0.1
nativescript-photoviewer: 2.1.5 (https://github.com/manijak/nativescript-photoviewer)

**Issue:**
When using the `nativescript-photoviewer` plugin in my app, right after `showGallery` the app crashes. This only happens if you have `nativescript-theme-core` plugin installed as well. 

**StackTrace:**
```
System.err: An uncaught Exception occurred on "main" thread.
System.err: Calling js method onGlobalLayout failed
System.err: TypeError: Cannot read property 'className' of undefined
System.err: 
System.err: StackTrace:
System.err:     Frame: function:'', file:'file:///node_modules/nativescript-theme-core/index.js:87:0
System.err:     Frame: function:'push.../node_modules/tns-core-modules/data/observable/observable.js.Observable.notify', file:'file:///node_modules/tns-core-modules/data/observable/observable.js:110:0
System.err:     Frame: function:'onGlobalLayout', file:'file:///node_modules/tns-core-modules/application/application.js:187:0
System.err:     at com.tns.Runtime.callJSMethodNative(Native Method)
System.err:     at com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1242)
System.err:     at com.tns.Runtime.callJSMethodImpl(Runtime.java:1122)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:1109)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:1089)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:1081)
System.err:     at com.tns.gen.android.view.ViewTreeObserver_OnGlobalLayoutListener.onGlobalLayout(ViewTreeObserver_OnGlobalLayoutListener.java:18)
System.err:     at android.view.ViewTreeObserver.dispatchOnGlobalLayout(ViewTreeObserver.java:945)
System.err:     at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2357)
System.err:     at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1460)
System.err:     at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7183)
System.err:     at android.view.Choreographer$CallbackRecord.run(Choreographer.java:949)
System.err:     at android.view.Choreographer.doCallbacks(Choreographer.java:761)
System.err:     at android.view.Choreographer.doFrame(Choreographer.java:696)
System.err:     at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:935)
System.err:     at android.os.Handler.handleCallback(Handler.java:873)
System.err:     at android.os.Handler.dispatchMessage(Handler.java:99)
System.err:     at android.os.Looper.loop(Looper.java:193)
System.err:     at android.app.ActivityThread.main(ActivityThread.java:6669)
System.err:     at java.lang.reflect.Method.invoke(Native Method)
System.err:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
JS:  *** StackTrace *** :       Frame: function:'', file:'file:///node_modules/nativescript-theme-core/index.js:87:0
JS:     Frame: function:'push.../node_modules/tns-core-modules/data/observable/observable.js.Observable.notify', file:'file:///node_modules/tns-core-modules/data/observable/observable.js:110:0
JS:     Frame: function:'onGlobalLayout', file:'file:///node_modules/tns-core-modules/application/application.js:187:0
JS:     at com.tns.Runtime.callJSMethodNative(Native Method)
JS:     at com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1242)
JS:     at com.tns.Runtime.callJSMethodImpl(Runtime.java:1122)
JS:     at com.tns.Runtime.callJSMethod(Runtime.java:1109)
JS:     at com.tns.Runtime.callJSMethod(Runtime.java:1089)
JS:     at com.tns.Runtime.callJSMethod(Runtime.java:1081)
JS:     at com.tns.gen.android.view.ViewTreeObserver_OnGlobalLayoutListener.onGlobalLayout(ViewTreeObserver_OnGlobalLayoutListener.java:18)
JS:     at android.view.ViewTreeObserver.dispatchOnGlobalLayout(ViewTreeObserver.java:945)
JS:     at android.view.ViewRootImpl....
```

**Workaround:**
Updated `index.js` of `nativescript-theme-core`:
```js
on(displayedEvent, () => {
    const root = getRootView();
    if(!root) return; // Root was undefined when launching photoviewer...
 
    const classList = new ClassList(root.className);
    classList.add("ns-root", `ns-${isAndroid ? "android" : "ios"}`, `ns-${device.deviceType.toLowerCase()}`);

    root.className = classList.get();

    if (!started) {
        handleOrientation({ newValue: getOrientation() });
        on(orientationChangedEvent, handleOrientation);
        started = true;
    }
});
```

**Comment by bundyo** [comment]
Thanks for the bug report, I'll check it out. Strange that the root view is missing there.

**Comment by smorcuend** [comment]
Same issue with nativescript-barcodescanner: https://github.com/EddyVerbruggen/nativescript-barcodescanner.

Same behaviour on scan({...}) method

nativescript: 6.0.3
tns-core-modules: 6.0.7
theme-core: 2.0.9
platform-android: 6.0.2
nativescript-barcodescanner: 3.0.2

Stack trace:
```System.err: An uncaught Exception occurred on "main" thread.
System.err: Calling js method onGlobalLayout failed
System.err: TypeError: Cannot read property 'className' of undefined
System.err: 
System.err: StackTrace:
System.err: 	Frame: function:'', file:'file:///node_modules/nativescript-theme-core/index.js:114:0
System.err: 	Frame: function:'push.../node_modules/tns-core-modules/data/observable/observable.js.Observable.notify', file:'file:///node_modules/tns-core-modules/data/observable/observable.js:110:0
System.err: 	Frame: function:'onGlobalLayout', file:'file:///node_modules/tns-core-modules/application/application.js:187:0
System.err: 	at com.tns.Runtime.callJSMethodNative(Native Method)
System.err: 	at com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1242)
System.err: 	at com.tns.Runtime.callJSMethodImpl(Runtime.java:1122)
System.err: 	at com.tns.Runtime.callJSMethod(Runtime.java:1109)
System.err: 	at com.tns.Runtime.callJSMethod(Runtime.java:1089)
System.err: 	at com.tns.Runtime.callJSMethod(Runtime.java:1081)
System.err: 	at com.tns.gen.android.view.ViewTreeObserver_OnGlobalLayoutListener.onGlobalLayout(ViewTreeObserver_OnGlobalLayoutListener.java:18)
System.err: 	at android.view.ViewTreeObserver.dispatchOnGlobalLayout(ViewTreeObserver.java:945)
System.err: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2238)
System.err: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1386)
System.err: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6733)
System.err: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
System.err: 	at android.view.Choreographer.doCallbacks(Choreographer.java:723)
System.err: 	at android.view.Choreographer.doFrame(Choreographer.java:658)
System.err: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
System.err: 	at android.os.Handler.handleCallback(Handler.java:789)
System.err: 	at android.os.Handler.dispatchMessage(Handler.java:98)
System.err: 	at android.os.Looper.loop(Looper.java:164)
System.err: 	at android.app.ActivityThread.main(ActivityThread.java:6541)
System.err: 	at java.lang.reflect.Method.invoke(Native Method)
System.err: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
System.err: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)

**Comment by smorcuend** [comment]
Thanks @manijak for these workaround, it works perfectly!

**Comment by bundyo** [comment]
Yeah, though with 6.1 these parts of the theme will become obsolete and will be handled by core modules.

**Comment by bundyo** [comment]
Fixed in 2.0.10.
